data-service/src/test/java/com/amazonaws/blox/dataservice/handler/EnvironmentHandlerTest.java

```diff
@@ -0,0 +1,338 @@
+    when(environmentRepository.createEnvironment(isA(Environment.class)))
+        .thenThrow(new StorageException(""));
+
+    environmentHandler.createEnvironment(createEnvironment);
+  }
+
+  @Test
+  public void createEnvironment()
+      throws EnvironmentExistsException, ServiceException, StorageException {
+    final Environment expectedEnvironment = createEnvironment;
+    expectedEnvironment.setHealth(EnvironmentHealth.Healthy);
+    expectedEnvironment.setStatus(EnvironmentStatus.Inactive);
+
+    when(environmentRepository.createEnvironment(isA(Environment.class)))
+        .thenReturn(expectedEnvironment);
+
+    final Environment result = environmentHandler.createEnvironment(createEnvironment);
+    assertNotNull(result.getEnvironmentVersion());
+    assertNotNull(result.getCreatedTime());
+
+    assertEquals(expectedEnvironment.getEnvironmentId(), createEnvironment.getEnvironmentId());
+    assertEquals(expectedEnvironment.getEnvironmentName(), createEnvironment.getEnvironmentName());
+    assertEquals(expectedEnvironment.getTaskDefinition(), createEnvironment.getTaskDefinition());
+    assertEquals(expectedEnvironment.getRole(), createEnvironment.getRole());
+    assertEquals(
+        expectedEnvironment.getInstanceGroup().getCluster(),
+        createEnvironment.getInstanceGroup().getCluster());
+    assertEquals(expectedEnvironment.getType(), createEnvironment.getType());
+    assertEquals(expectedEnvironment.getHealth(), createEnvironment.getHealth());
+    assertEquals(expectedEnvironment.getStatus(), createEnvironment.getStatus());
+  }
+
+  @Test(expected = NullPointerException.class)
+  public void createEnvironmentTargetVersionNullEnvironmentId()
+      throws EnvironmentNotFoundException, EnvironmentExistsException, ServiceException {
+    environmentHandler.createEnvironmentTargetVersion(null, ENVIRONMENT_VERSION);
+  }
+
+  @Test(expected = NullPointerException.class)
+  public void createEnvironmentTargetVersionNullEnvironmentVersion()
+      throws EnvironmentNotFoundException, EnvironmentExistsException, ServiceException {
+    environmentHandler.createEnvironmentTargetVersion(ENVIRONMENT_ID, null);
+  }
+
+  @Test(expected = ServiceException.class)
+  public void createEnvironmentTargetVersionGetFails()
+      throws EnvironmentNotFoundException, EnvironmentExistsException, ServiceException,
+          StorageException {
+    when(environmentRepository.getEnvironment(ENVIRONMENT_ID, ENVIRONMENT_VERSION))
+        .thenThrow(new StorageException(""));
+    environmentHandler.createEnvironmentTargetVersion(ENVIRONMENT_ID, ENVIRONMENT_VERSION);
+  }
+
+  @Test(expected = EnvironmentNotFoundException.class)
+  public void createEnvironmentTargetVersionGetReturnsNull()
+      throws EnvironmentNotFoundException, EnvironmentExistsException, ServiceException,
+          StorageException {
+    when(environmentRepository.getEnvironment(ENVIRONMENT_ID, ENVIRONMENT_VERSION))
+        .thenReturn(null);
```

Comment:
Is this how the `getEnvironment` method is defined to behave? I couldn't find a test for `repo.getEnvironment()` that tests the case where the environment is not found.

Reply:
DynamodbMapper.load returns null if the record is not found